### ISSUE TITLE
"a!=b"  does not return a boolean value

### ISSUE THREAD

**Issue by JC-LL** [crystal-lang/crystal#5663]
I found a very strange bug in Crystal : my simple Euclid GCD does not return ! Apparently, the comparison operation "a!=b"  does not return a boolean value, but the last operand (b) ! 

Crystal 0.23.1 [e2a1389] (2017-07-13) LLVM 3.8.1

```

    def gcd(a,b)
      while a!=b
        puts "a=#{a} b=#{b} diff? #{a!=b}"
        if a>b
          a-=b
        else
          b-=a
        end
      end
      return a
    end
    
    a,b=124,234
    res=gcd(a,b)
    puts "gcd(#{a},#{b})=#{res}"
```

**Comment by jhass** [comment]
This appears to be fixed in 0.24.1, compare and https://carc.in/#/r/3hkg (0.23.1) and https://carc.in/#/r/3hkf (0.24.1)

**Comment by asterite** [comment]
To explain why it happens, `a!=b` was being parsed as `a! = b`.

For better clarity, try to use spaces between operands :-)
